testsuite/domain/src/test/java/org/jboss/as/test/integration/domain/AbstractSlaveHCAuthenticationTestCase.java

```diff
@@ -0,0 +1,115 @@
+
+import static org.jboss.as.controller.descriptions.ModelDescriptionConstants.ADMIN_ONLY;
+import static org.jboss.as.controller.descriptions.ModelDescriptionConstants.CORE_SERVICE;
+import static org.jboss.as.controller.descriptions.ModelDescriptionConstants.HOST;
+import static org.jboss.as.controller.descriptions.ModelDescriptionConstants.HOST_STATE;
+import static org.jboss.as.controller.descriptions.ModelDescriptionConstants.MANAGEMENT;
+import static org.jboss.as.controller.descriptions.ModelDescriptionConstants.NAME;
+import static org.jboss.as.controller.descriptions.ModelDescriptionConstants.OP;
+import static org.jboss.as.controller.descriptions.ModelDescriptionConstants.OP_ADDR;
+import static org.jboss.as.controller.descriptions.ModelDescriptionConstants.OUTCOME;
+import static org.jboss.as.controller.descriptions.ModelDescriptionConstants.READ_ATTRIBUTE_OPERATION;
+import static org.jboss.as.controller.descriptions.ModelDescriptionConstants.RESULT;
+import static org.jboss.as.controller.descriptions.ModelDescriptionConstants.SECRET;
+import static org.jboss.as.controller.descriptions.ModelDescriptionConstants.SECURITY_REALM;
+import static org.jboss.as.controller.descriptions.ModelDescriptionConstants.SERVER_IDENTITY;
+import static org.jboss.as.controller.descriptions.ModelDescriptionConstants.SUCCESS;
+import static org.jboss.as.controller.descriptions.ModelDescriptionConstants.VALUE;
+import static org.jboss.as.controller.descriptions.ModelDescriptionConstants.WRITE_ATTRIBUTE_OPERATION;
+
+import java.io.IOException;
+import java.util.concurrent.CancellationException;
+import java.util.concurrent.ExecutionException;
+
+import org.jboss.as.controller.client.ModelControllerClient;
+import org.jboss.as.controller.client.OperationBuilder;
+import org.jboss.dmr.ModelNode;
+import org.junit.Assert;
+
+/**
+ * Encapsulates some of the logic for tests where a slave host controller attempts to authenticate to master controller.
+ */
+public abstract class AbstractSlaveHCAuthenticationTestCase {
+
+    private static final int TIMEOUT = 60000;
+
+    protected abstract ModelControllerClient getDomainMasterClient();
+    protected abstract ModelControllerClient getDomainSlaveClient();
+
+    protected void reloadSlave() throws Exception {
+        ModelNode op = new ModelNode();
+        op.get(OP).set("reload");
+        op.get(OP_ADDR).add(HOST, "slave");
+        op.get(ADMIN_ONLY).set(false);
+        try {
+            getDomainSlaveClient().execute(new OperationBuilder(op).build());
+        } catch(IOException e) {
+            final Throwable cause = e.getCause();
+            if (!(cause instanceof ExecutionException) && !(cause instanceof CancellationException)) {
+                throw e;
+            } // else ignore, this might happen if the channel gets closed before we got the response
+        }
+
+        // Wait until host is reloaded
+        readHostControllerStatus(getDomainSlaveClient(), TIMEOUT);
```

Comment:
Because the code is moving. ;) Its a good point, though I'll take a look at these.